src/d3a/d3a_core/sim_results/device_statistics.py

```diff
@@ -125,12 +125,13 @@ def _load_profile_stats(cls, area: Area, subdict: Dict):
 
     def update(self, area):
         self.gather_device_statistics(area, self.device_stats_dict, self.flat_results_dict)
-        self.flat_stats_time_str = convert_datetime_to_str_keys(self.flat_results_dict, {})
         self.device_stats_time_str = convert_datetime_to_str_keys(self.device_stats_dict, {})
 
         # Calculate last market stats
         self.gather_device_statistics(area, {}, self.current_stats_dict)
         self.current_stats_time_str = convert_datetime_to_str_keys(self.current_stats_dict, {})
+        merge_device_statistics_results_to_global(
+            self.current_stats_time_str, self.flat_stats_time_str)
```

Comment:
AFAIU, flat_stats_time_str will be an empty dict in any case, because it is not populated any more. Is this on purpose, or did you really wanted to parse `flat_results_dict` here?

Reply:
You could also read line 128, right?

Reply:
Line 127, but indeed :) flat_results_dict is still populated and used.